force-app/main/selector/DonationHistorySelector.cls

```diff
@@ -38,39 +38,58 @@ public inherited sharing class DonationHistorySelector {
     * @date 2021
     * @description method to return OpportunityContactRole data.
     */
-    public List<OpportunityContactRole> getWonOpportunityContactRolesByContactId(Id contactId) {
+    public List<Opportunity> getWonOpportunitiesByContactId(Id contactId) {
         return [
             SELECT Id, 
-                Opportunity.Name, 
-                Opportunity.CloseDate, 
-                Opportunity.Amount, 
-                Contact.Email, 
-                Contact.Name, 
-                Role, 
-                IsPrimary
-            FROM OpportunityContactRole 
-            WHERE Opportunity.IsWon = true 
-            AND ContactId =:contactId 
+                Name, 
+                CloseDate, 
+                Amount, 
+                Primary_Contact__r.Email, 
+                Primary_Contact__r.Name
+            FROM Opportunity
+            WHERE IsWon = true 
+            AND Primary_Contact__r.Id =:contactId 
             WITH SECURITY_ENFORCED
-            ORDER BY Opportunity.CloseDate 
-            DESC LIMIT 2000
+            ORDER BY CloseDate 
+            DESC LIMIT 50
         ];
     }
-    /** 
-    * @author Salesforce.org
-    * @date 2021
-    * @description method to return Partial_Soft_Credit__c data.
-    */
-    public List<Partial_Soft_Credit__c> getPartialSoftCredits(Set<Id> opportunityContactRoleIds) {
+
+    /**
+     * @description Returns the OpportunityContactRole won in a specific year.
+     * @param contactId the contact id.
+     * @param year Integer filter of year.
+     * @return List<OpportunityContactRole>
+     */
+    public List<Opportunity> getWonOpportunitiesByContactIdAndYear(Id contactId, Integer year) {
         return [
-            SELECT 
-                Id, 
-                Amount__c, 
-                Role_Name__c,
-                Contact_Role_ID__c 
-            FROM Partial_Soft_Credit__c 
-            WHERE Contact_Role_ID__c IN :opportunityContactRoleIds
+            SELECT  Id, 
+                    Name, 
+                    CloseDate, 
+                    Amount, 
+                    Primary_Contact__r.Email, 
+                    Primary_Contact__r.Name
+            FROM Opportunity
+            WHERE IsWon = true 
+                AND Primary_Contact__r.Id =:contactId 
+                AND Calendar_Year(CloseDate) = :year
             WITH SECURITY_ENFORCED
+            ORDER BY Opportunity.CloseDate 
+            DESC LIMIT 2000
         ];
-    }   
+    }
+
+    /**
+     * @description Returns the years in which the contact has made at least 1 donation.
+     * @param contactId Id the contact id.
+     * @returns List<AggregateResult> Selecting the Calenday_Year of the Opportunity.CloseDate
+     */
+    public List<AggregateResult> getYearsWithDonationForContact(Id contactId) {
+        return [SELECT   Calendar_Year(CloseDate)
```

Comment:
See suggestion in DonationHistoryService around giving this aggregate expression a name so we don't have to use `'expr0'`.

```suggestion
        return [SELECT   Calendar_Year(CloseDate) closeYear
```